lib/connection.cpp

```diff
@@ -539,8 +539,8 @@ void Connection::loginWithToken(const QByteArray& loginToken,
 {
     Q_ASSERT(d->data->baseUrl().isValid() && d->loginFlows.contains(LoginFlows::Token));
     d->loginToServer(LoginFlows::Token.type,
-                     none /*user is encoded in loginToken*/, "" /*password*/,
-                     loginToken, deviceId, initialDeviceName);
+                     none /*user is encoded in loginToken*/, QString() /*password*/,
+                     QString::fromUtf8(loginToken), deviceId, initialDeviceName);
```

Comment:
```suggestion
                     QString::fromLatin1(loginToken), deviceId, initialDeviceName);
```
But in fact we have to switch `loginToken` back to `QString` because the SSO code calls this after retrieving the token as `QString` (calling `toLatin1()` to accommodate).